examples/encoding/plot_continuous_encoder.py

```diff
@@ -0,0 +1,114 @@
+inputs. In this case, a encoding model is fit for one EEG channel, using the
+values of all other channels as inputs. The coefficients in this case are
+interpreted as a measure of functional connectivity. These inputs could also
+be stimulus features, such as spectral features of sound. In this case, model
+coefficients would be measures of receptive field properties for the channel.
+
+Estimation of spectro-temporal and spatio-temporal receptive fields using
+modeling with continuous inputs is described in:
+
+    Theunissen, F. E. et al. Estimating spatio-temporal receptive
+    fields of auditory and visual neurons from their responses to
+    natural stimuli. Network 12, 289-316 (2001).
+
+    Willmore, B. & Smyth, D. Methods for first-order kernel
+    estimation: simple-cell receptive fields from responses to
+    natural scenes. Network 14, 553-77 (2003).
+"""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+import matplotlib.pyplot as plt
+
+import mne
+import numpy as np
+from mne.datasets import sample
+from mne.encoding import SubsetEstimator, FeatureDelayer, get_coefs
+from mne.decoding import Vectorizer
+from sklearn.pipeline import make_pipeline
+from sklearn.linear_model import Ridge
+from sklearn.preprocessing import StandardScaler
+
+###############################################################################
+# Load and preprocess data
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+raw = mne.io.read_raw_fif(raw_fname, preload=True).pick_types(
+    meg=False, stim=False, eeg=True).filter(1, None, method='iir')
+seed_channel = 'EEG 011'  # Choosing a frontal electrode
+ix_seed = mne.pick_channels(raw.ch_names, [seed_channel])[0]
+ix_targets = np.setdiff1d(range(len(raw.ch_names)), [ix_seed])
+
+# Set up encoding model variables
+# Multiple continuous electrode values as input
+X = raw[:][0][ix_targets].T  # estimators expect shape (n_samples, n_features)
+# Single continuous electrode values as output
+y = raw[:][0][ix_seed]
+n_times, n_targets = X.shape
+
+###############################################################################
+# Fit the model / Make predictions
+# Define model preprocessing and parameters
+ix_train = np.arange(n_times // 2)
+ix_test = np.arange(n_times // 2, n_times)
+
+# This creates time-lagged versions of the data (in seconds)
+delay_step = .01  # Time between delays (in seconds)
+delayer = FeatureDelayer(delays=np.arange(-.1, .1, delay_step),
+                         sfreq=raw.info['sfreq'])
+info_new = raw.copy().drop_channels([seed_channel]).info.copy()
+info_new['sfreq'] = 1. / delay_step
+
+# This is the final estimator that will be called after masking data
+estimator = make_pipeline(StandardScaler(), Ridge(alpha=1.))
+# This allows us to use a subset of indices for training / prediction
+masker = SubsetEstimator(estimator, samples_train=ix_train,
+                         samples_pred=ix_test)
```

Comment:
I don't think we should init an estimator with something that varies with `n_samples`.

What do you want to keep `samples_train` and `samples_pred` in the first place? How about passing a cv (or cv constructor)?